fog/view/server/src/db_fetcher.rs

```diff
@@ -282,64 +291,67 @@ impl<DB: RecoveryDb + Clone + Send + Sync + 'static> DbFetcherThread<DB> {
-                    self.block_tracker.block_processed(ingress_key, block_index);
-
-                    // Store the fetched records so that they could be consumed by the enclave
-                    // when its ready.
-                    {
-                        let mut state = self.shared_state();
-                        state.fetched_records.push(FetchedRecords {
+                Ok(block_results) => {
+                    if !block_results.is_empty() {
+                        // Log
+                        log::info!(
+                            self.logger,
+                            "ingress_key {:?} fetched {} blocks starting with block {}",
                             ingress_key,
+                            block_results.len(),
                             block_index,
-                            records: tx_outs,
-                        });
+                        );
+
+                        if block_results.len() == self.block_batch_request_size {
+                            // Ingest has produced as much block data as we asked for,
+                            // we'd like to keep trying to download in the next loop iteration.
+                            may_have_more_work = true;
+                        }
+
+                        for (idx, tx_outs) in block_results.into_iter().enumerate() {
+                            let idx = idx as u64;
+                            let num_tx_outs = tx_outs.len();
+
+                            // Mark that we are done fetching data for this block.
+                            self.block_tracker
+                                .block_processed(ingress_key, block_index + idx);
+
+                            // Store the fetched records so that they could be consumed by the
+                            // enclave when its ready.
+                            {
+                                let mut state = self.shared_state();
+                                state.fetched_records.push(FetchedRecords {
+                                    ingress_key,
+                                    block_index: block_index + idx,
+                                    records: tx_outs,
+                                });
+                            }
+
+                            // Update metrics.
+                            counters::BLOCKS_FETCHED_COUNT.inc();
+                            counters::TXOS_FETCHED_COUNT.inc_by(num_tx_outs as u64);
+
+                            // Block if we have queued up enough records for now.
+                            // (Until the enclave thread drains the queue).
+                            let (lock, condvar) = &*self.num_queued_records_limiter;
+                            let mut num_queued_records = condvar
+                                .wait_while(lock.lock().unwrap(), |num_queued_records| {
+                                    *num_queued_records > MAX_QUEUED_RECORDS
+                                })
```

Comment:
Nit: Unnecessary block
```suggestion
                                .wait_while(
                                    lock.lock().unwrap(),
                                    |num_queued_records| *num_queued_records > MAX_QUEUED_RECORDS
                                )
```

Reply:
cargo fmt put these back for some reason

:shrug: